Source/Core/Core/PowerPC/PPCSymbolDB.cpp

```diff
@@ -399,6 +402,61 @@ bool PPCSymbolDB::LoadMap(const Core::CPUThreadGuard& guard, const std::string&
     if (name[strlen(name) - 1] == '\r')
       name[strlen(name) - 1] = 0;
 
+    // Split the current name string into separate parts, and get the object name
+    // if it exists.
+    std::string processed_name = TabsToSpaces(0, name);  // Remove tabs
+    std::vector<std::string> parts = SplitString(processed_name, ' ');
+    size_t num_parts = parts.size();
+
+    std::string object_filename_string = "";
+    std::string name_string = name;  // Default to the full line
+
+    if (num_parts > 0)
+    {
+      // If the first part does not contain a left bracket (not a demangled symbol), use the first
+      // part
+      size_t parenthesis_offset = parts[0].find('(');
+      if (parenthesis_offset == std::string::npos)
+      {
+        name_string = parts[0];
+      }
+      else
+      {
+        // If it does, the symbol is likely demangled, so look for the end of the name
+        size_t name_end_offset = parenthesis_offset;
+        int depth = 0;  // Current parenthesis depth
+        while (true)
+        {
+          char cur_char = processed_name[name_end_offset];
+          if (cur_char == '(')
+            depth++;
+          else if (cur_char == ')')
+            depth--;
+
+          name_end_offset++;
+
+          if (depth == 0)
+            break;
+        }
+
+        name_end_offset++;  // Advance past the ending parenthesis
+
+        // If the name ends with const, advance past it
+        if (processed_name.find(") const") != std::string::npos)
+        {
+          name_end_offset += 6;
+        }
+
+        name_string = processed_name.substr(0, name_end_offset);
+      }
+
+      // If the last part contains a ., it has to be the object name
+      if (num_parts > 1 && parts[num_parts - 1].find('.') != std::string::npos)
+      {
+        object_filename_string = parts[num_parts - 1];
+      }
+    }
+
```

Comment:
This approach is way too complex and have several flaws which can crash Dolphin and clobber some symbol maps:
 - methods not only can be marked as `const`, but also `volatile`, `noexcept`, ... _(though, I haven't come cross these yet)_
 - there are cases when the object part doesn't have a `.` character.
 - some symbols have several objects separated with spaces.
 - it crashes Dolphin when the number of open/close parentheses don't match (i.e. overflow in the while true loop).

Using other approaches such as: 
 - using the tab as delimiter _(for simpler cases)_
 - parsing the string by starting at the end and picking each string until reaching the function name or a parenthesis might work better _(this approach might fail if the method has cv qualifier)_.

Reply:
Addressing your concerns:
-The only unmangled map (Twilight Princess Wii) only features const at the end, so the fact there can be volatile/noexcept/etc... is a non issue IMO. Ideally anybody using the custom maps left on the disc will switch over to a reformatted symbol map, and this superfluous handling code can be phased out eventually, which I was told by someone in the Twilight Princess speedrunning/reversing engineering communities is something that would be desired/possible. But regardless, I feel that this suffices.
-What cases? All standard symbol maps that would be supported by Dolphin (Codewarrior) always have an object name with a dot as far as I know. If you could give me counterexamples on Discord I would appreciate it however.
-From testing with the Wii TP map I never found any cases of it entering an infinite loop, but regardless I have added string bounds checking.

And yes, I agree it's complex, but if I cannot use regex, and have to deal with a non-standard map format then I don't have much of a choice. I don't think that parsing in reverse is a good idea, it would just make things more complicated IMO. However, I am willing to give splitting on the tab a try.